### PR TITLE
samples: net: sockets: echo:  update socket offload after net logging overhaul, update sample.yaml

### PR DESCRIPTION
**Pull request by GAnthony**
This patch gets the socket offload module and sockets echo sample to build after the net logging overhaul.

Also, adds updates sample.yaml to build for wifi based drivers (including SimpleLink).

### CODE REVIEW

File: samples/net/wifi/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+sample:
```
Comment (jukkar) [inline samples/net/wifi/sample.yaml]: I did not notice that you had added sample.yaml file for wifi, so I have similar addition for that in #10405

**Comment by pfalcon** [comment]
Commit message:

> wifi samples (sockets/echo, net/wifi)

Can you please replace with "wifi-enabled samples" to avoid confusion.

And you'll need to decide with @jukkar whose sample.yaml is better.

File: samples/net/wifi/sample.yaml (hunk 1 above)
Comment (GAnthony) [reply samples/net/wifi/sample.yaml]: OK, then I'll remove this one.

**Comment by GAnthony** [comment]
OK, I removed my sample/net/wifi/sample.yaml, and also got socket offload and sockets/echo sample to build again with new logging updates.

File: drivers/wifi/simplelink/simplelink_sockets.c [hunk 2]
```diff
@@ -4,6 +4,10 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define LOG_LEVEL CONFIG_WIFI_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(net_socket_offload);
```
Comment (jukkar) [inline drivers/wifi/simplelink/simplelink_sockets.c]: To my understanding, the module name must be unique but you are using the same name in socket_offload.c. Better to call this module "net_wifi_socket_offload".

Comment (GAnthony) [reply drivers/wifi/simplelink/simplelink_sockets.c]: I noticed that each call to LOG_MODULE_REGISTER() is adding a new constant struct in memory for that file.
Reading logger.rst, and looking at how power module is using logging, it seemed a good opportunity to explore LOG_MODULE_DECLARE(), where the module name is registered only once, and other files which are part of the "module" only need to declare usage.
Here, I take the liberty of categorizing the "simplelink wifi driver" as a "module."
So, I'll push an update to this effect, and fix the shippable complaint on commit line length.

File: drivers/wifi/simplelink/simplelink_support.c [hunk 3]
```diff
@@ -218,7 +218,7 @@ void SimpleLinkWlanEventHandler(SlWlanEvent_t *wlan_event)
 		memcpy(sl_conn.bssid, wlan_event->Data.Connect.Bssid,
 		       BSSID_LEN_MAX);
 
-		LOG_INF("\n[WLAN EVENT] STA Connected to the AP: %s, "
+		LOG_INF("\n\r[WLAN EVENT] STA Connected to the AP: %s, "
```
Comment (pfalcon) [inline drivers/wifi/simplelink/simplelink_support.c]: I believe even when it was initially introduced, there was a comment that it's a bad thing, and it now pays off. So, it's another chance to remove that \n altogether, there's a simple rule to follow: one logging action - one log line produced.

If that doesn't work, then at lease make it \r\n, as the standard patter for it goes.

**Review by pfalcon — CHANGES_REQUESTED** [review]
Needs some cleanup.

File: drivers/wifi/simplelink/simplelink_support.c (hunk 3 above)
Comment (jukkar) [reply drivers/wifi/simplelink/simplelink_support.c]: > \r\n, as the standard patter for it goes.

What standard is that? We should not add these \r, that is totally pointless thing to do. See also #10399 

I also think that adding \n to logging strings is useless as the output might go to some backend where \n makes no sense.

File: drivers/wifi/simplelink/simplelink_sockets.c (hunk 2 above)
Comment (jukkar) [reply drivers/wifi/simplelink/simplelink_sockets.c]: > Here, I take the liberty of categorizing the "simplelink wifi driver" as a "module."

Indeed, makes sense.

File: drivers/wifi/simplelink/simplelink_support.c (hunk 3 above)
Comment (GAnthony) [reply drivers/wifi/simplelink/simplelink_support.c]: OK, will (finally) remove those \n 's.

Comment (pfalcon) [reply drivers/wifi/simplelink/simplelink_support.c]: :+1:

**Comment by GAnthony** [comment]
OK, feedback resolved, and rebased.

File: drivers/wifi/simplelink/simplelink_log.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/**
```
Comment (pfalcon) [inline drivers/wifi/simplelink/simplelink_log.h]: I don't remember this file in the previous PR version ;-). Let me approve, but IMHO, it's superfluous.

**Review by pfalcon — APPROVED** [review]

File: drivers/wifi/simplelink/simplelink_log.h (hunk 4 above)
Comment (pfalcon) [reply drivers/wifi/simplelink/simplelink_log.h]: 2 lines of defines specifically. It does so using 23 lines. And it takes 1 line to include it. And it's included 3 times. For me, the math doesn't even out, but that's just me ;-).

**Review by jukkar — APPROVED** [review]

**Pull request merged**